lib/catalog/rbac/access.rb

```diff
@@ -30,21 +30,24 @@ def admin_check
       end
 
       def resource_check(verb, id = @user.params[:id], klass = @user.controller_name.classify.constantize)
-        return unless rbac_enabled?
-        return if catalog_admin?
+        return true unless rbac_enabled?
+        return true if catalog_admin?
 
+        return false unless access_object(@user.controller_name.classify.constantize.table_name, verb).accessible?
         ids = access_id_list(verb, klass)
         if klass.try(:supports_access_control?)
-          raise Catalog::NotAuthorized, "#{verb.titleize} access not authorized for #{klass}" if ids.exclude?(id.to_s)
+          return false if ids.exclude?(id.to_s)
         end
+
+        return true
```

Comment:
Same

Reply:
Same reason as above, but for the policy definitions that use `resource_check`.